app/controllers/books_controller.rb

```diff
@@ -4,18 +4,20 @@ class BooksController < ApplicationController
   def show
     @book = Book.find(params[:id])
     authorize @book # authorize @book for pundit
-
-    if params[:district].nil?
-      @search_district = ""
-    else
-      @search_district = params[:district]
-    end
     # get users that have the selected book
     @book_users = @book.users
     # match book users with selected district, e. g. "Neukölln"
     @locations = []
-    @book_users.each do |book_user|
-      @locations += book_user.locations.select { |location| location.district == @search_district }
+    # check if params[:district] was passed
+    if params[:district].nil?
+      @book_users.each do |book_user|
+        @locations += book_user.locations.select { |location| location }
```

Comment:
This looks odd to me. I think the select statement does nothing here. 
`@locations += book_user.locations` would give the exact same result.

Reply:
True, I fixed it.